examples/webgl/src/main.rs

```diff
@@ -45,45 +34,37 @@ impl Component for App {
         }
     }
 
-    fn rendered(&mut self, ctx: &Context<Self>, first_render: bool) {
+    fn rendered(&mut self, ctx: &Context<Self>, _first_render: bool) {
         // Once rendered, store references for the canvas and GL context. These can be used for
         // resizing the rendering area when the window or canvas element are resized, as well as
         // for making GL calls.
-
         let canvas = self.node_ref.cast::<HtmlCanvasElement>().unwrap();
-
         let gl: GL = canvas
             .get_context("webgl")
             .unwrap()
             .unwrap()
             .dyn_into()
             .unwrap();
-
-        self.gl = Some(gl);
-
-        // In a more complex use-case, there will be additional WebGL initialization that should be
-        // done here, such as enabling or disabling depth testing, depth functions, face
-        // culling etc.
-
-        if first_render {
-            // The callback to request animation frame is passed a time value which can be used for
-            // rendering motion independent of the framerate which may vary.
-            let handle = {
-                let link = ctx.link().clone();
-                request_animation_frame(move |time| link.send_message(Msg::Render(time)))
-            };
-
-            // A reference to the handle must be stored, otherwise it is dropped and the render
-            // won't occur.
-            self._render_loop = Some(handle);
-        }
+        self.gl = Some(Rc::new(gl));
+        self.render_gl(ctx.link());
```

Comment:
Nit*: `if first_render` would be more correct here. Yew allows itself to (and with SSR actually performs) spurious re-renders, so that would spawn multiple render loops. This is compounded by the fact that the current approach doesn't offer a way to cancel the running loop (and the draw function Rc leaks by capturing a ref to itself in the closure inside).

* this is mostly a correctness concern. 'In practice' the current approach works fine, cause Props never get updated, no messages cause re-renders and it's not used in an SSR context.